src/components/data-exploration/hierarchical-tree/HierarchicalTree.jsx

```diff
@@ -50,24 +50,27 @@ const HierarchicalTree = (props) => {
     // If rootNode, ignore
     if (dragNode.rootNode) return;
 
+    if (dropPosition === -1) return;
+
     // pos is a string e.g.: 0-0-1, each number is a position in a tree level
     const posFromArray = dragNode.pos.split('-');
     const posToArray = node.pos.split('-');
 
-    const fromPosition = parseInt(posFromArray[2], 10);
+    const sameLevel = (posFromArray.length === posToArray.length);
 
     // If not in the same cellClass, ignore
     if (!_.isEqual(posFromArray[1], posToArray[1])) return;
 
-    // If was dropped in same place, ignore
-    if (fromPosition === dropPosition) return;
+    const numberOfClusters = treeData[posFromArray[1]].children.length;
 
-    // If not dropped in gap, ignore
-    // (only allow dropToGap when the destination node is rootNode
-    // because it can have children nodes)
-    if (!dropToGap && !node.rootNode) return;
+    const fromPosition = parseInt(posFromArray[2], 10);
+    const toPosition = (!sameLevel ? dropToGap ? numberOfClusters : 0 : dropPosition);
```

Comment:
Can you rewrite this line so that it's simpler to undrstand?

Reply:
I rewrote it and added a comment. For one of the possible drop positions under the last cluster in a list, the `dropPosition` is not set correctly. This line is used to set the position to the last position (equivalent to the number of clusters), the condition (rewritten as `!sameLevel && dropToGap`) is based on the properties of the drop position (leftmost drop position under Cluster 8  in the diagram in the ticket).

Reply:
`dropPosition` for the first position is also not set correctly. So the double condition is needed to set `toPosition` to `0` instead of using `dropPosition`.